src/main/java/picard/fingerprint/CrosscheckFingerprints.java

```diff
@@ -787,7 +797,9 @@ private int crossCheckFingerprints(final Map<FingerprintIdDetails, Fingerprint>
 
             for (int col = 0; col < rhsFingerprintIdDetails.size(); col++) {
                 final FingerprintIdDetails rhsId = rhsFingerprintIdDetails.get(col);
-                final boolean expectedToMatch = EXPECT_ALL_GROUPS_TO_MATCH || lhsId.sample.equals(rhsId.sample);
+                final String lhsMatchId = sampleIndividualMap == null ? lhsId.sample : sampleIndividualMap.getOrDefault(lhsId.sample, lhsId.sample);
```

Comment:
this should be a private method `resolveSampleIfPossible` (or somesuch)

Reply:
done